site/app/controllers/calendar/CalendarController.php

```diff
@@ -27,8 +31,125 @@ class CalendarController extends AbstractController {
      */
     public function viewCalendar(): WebResponse {
         $user = $this->core->getUser();
-        $gradeables_of_user = GradeableUtils::getAllGradeableListFromUserId($this->core, $user);
 
-        return new WebResponse(CalendarView::class, 'showCalendar', CalendarInfo::loadGradeableCalendarInfo($this->core, $gradeables_of_user));
+        $calendar_messages = [];
+
+        $courses = $this->core->getQueries()->getCourseForUserId($user->getId());
+
+        $gradeables_of_user = GradeableUtils::getAllGradeableListFromUserId($this->core, $user, $courses, $calendar_messages);
+
+        return new WebResponse(CalendarView::class, 'showCalendar', CalendarInfo::loadGradeableCalendarInfo($this->core, $gradeables_of_user, $courses, $calendar_messages));
+    }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/calendar")
+     * @return WebResponse
```

Comment:
This PHPDoc line is unnecessary since the return value already has a native PHP type declaration